Fix for the damn annoying lets mash all projects in a single one bug

smiffy6969 opened the pull request:
Fix for the damn annoying lets mash all projects in a single one bug where all your open files from various projects get mashed into one, stopping you from opening some projects sometimes as it keeps opening another directory tree.

smiffy6969 commented:
not sure how permanent this solution will be, I notice atom like to move things about a bit. Works for me though, I now have this indispensable extension back again...

mpeterson2 commented:
Is this supposed to solve #75? It doesn't seem to change anything when I try it. I still get my old folder to show up when I open a new window.

I think the issue here is that Atom changed the way that events are emitted so line 34 of `projects.coffee` doesn't work anymore. I updated it to this a while ago and I thought it worked at one point, but apparently not anymore...

I appreciate you looking into the bug for sure though.

smiffy6969 commented:
Can gaurentee you it works fine in atom 184 as I have ran it all day with multiple projects.....
The issue is that the project path returns undefined so I am grabbing the path from different object value.
Before this fix I could not use this extension, since the fix it is now working fine.
Go into your projects save folder in save session and you should see an undefined folder, delete that, add fix, close and open atom and open a project, save and you will see new folder created on save is now no longer undefined and working.
I tested this on 184 atom, not saying it is going to work are this or post this, that's not really my job. But it is defo broke against this version of atom and unusable, and this defo fixes it for me.
Paul
Sent using CloudMagic
On Wed, Mar 18, 2015 at 11:09 pm, Michael Peterson <notifications@github.com> wrote:Is this supposed to solve #75? It doesn't seem to change anything when I try it. I still get my old folder to show up when I open a new window.

I think the issue here is that Atom changed the way that events are emitted so line 34 of projects.coffee doesn't work anymore. I updated it to this a while ago and I thought it worked at one point, but apparently not anymore...

I appreciate you looking into the bug for sure though.

—Reply to this email directly or view it on GitHub.

smiffy6969 commented:
What version of atom are you running? this only became an issue since 184 (ubuntu 64 bit), so if you are testing atom lower than this the problem is not there.

I have just now updated chrome to 187... removed the save-session extension and re-installed, it is still broken in 187. I have added my fix again, and it is now working again....

I am not saying this is an issue in windows, because I don't run it, but in ubuntu (well ubuntu gnome), it is severely broken and unusable if you work on more than one project ever.

What happens is when you close your window, all session details are saved to a folder in home/name/.atom/packages/save-session/projects and the folder it saves the changes under is called 'undefined' because your extension cannot resolve the path name from atom.project.path

Instead I found the path of the project to be set at atom.project.rootDirectories which is an array, with the first value in the array being the correct path.

The problem is nothing to do with events being emitted, the issue is that when it tries to find the path to save the session details against, the path returns undefined as atom more than likely changed something in core or have broken something at there end (I see they have a habit of doing this, i think back to the workspace issue a while back and every package failing to grab workspace).

Quite simply this extension is unusable in ubuntu at anything past atom 184 unless I apply this fix, (restart twice) and all is good. Please can you take a look again, if not I may have to release a fork as I am not going to apply this fix every time you upgrade this extension, it gets a bit annoying.

If you can find a better fix by all means go for it, but its pretty clear the path is not being grabbed correctly when its creating folders called 'undefined' which is JS equivalent of 'Not Instantiated' followed up by testing to console.log the atom.project.path as 'undefined'.

mpeterson2 commented:
Okay, cool. I think I was looking at a different issue. If you could update the tests to pass then I can merge this in and update the package.

smiffy6969 commented:
Never done any testing on travis, no idea how to 'update tests to pass' 
and the travis error is a problem with curl (or at least thats what it 
says.....)

I have about ten projects on the go, 4 kids and no time to spend 3 hours 
working out how to get travis to pass this when you can manually merge 
in yourself.

Sorry mate but I have no time to learn travis at the moment.

Paul

On 19/03/15 17:35, Michael Peterson wrote:
>
> Okay, cool. I think I was looking at a different issue. If you could 
> update the tests to pass then I can merge this in and update the package.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/mpeterson2/save-session/pull/81#issuecomment-83689289>.
>

smiffy6969 commented:
write i have tweaked the fix for backwards compat (and to force travis to re test....) see what happens.

mpeterson2 commented:
There's no need to learn anything new. Travis testing is automatic when you push it to GitHub. All you have to do is update the tests in the [spec](https://github.com/mpeterson2/save-session/tree/master/spec) directory and will be good. I don't have much time for this project either, but it should be a small tweak so I'll look into it for you.

smiffy6969 commented:
never done any automated testing, I have updated test hoping it will fix it but i have no idea about auto testing scripts. When you spend all day writing commercial code you never get round to auto testing as no one wants to pay for it ;)

smiffy6969 commented:
shazammmmmm can you merge it now mate, added the project path tweak to tests and its passed.

mpeterson2 commented:
Awesome. Merging now.

smiffy6969 commented:
Cheers... Your a gent.... Gotta go make dinner now 
Sent using CloudMagic
On Thu, Mar 19, 2015 at 7:46 pm, Michael Peterson <notifications@github.com> wrote:Awesome. Merging now.

—Reply to this email directly or view it on GitHub.
